Invocation_ExceptionTest.test[2 - java.lang.OutOfMemoryError: message]

_master_ (commit 7815ff8a52642ac38b2e75aeaa6c77827f38085c)

Failed on ZuluJDK13 : http://jenkins.hazelcast.com/job/Hazelcast-3.x-ZuluJDK13/14/testReport/com.hazelcast.spi.impl.operationservice.impl/Invocation_ExceptionTest/test_2___java_lang_OutOfMemoryError__message_/

**Stacktrace**
```
java.lang.AssertionError: 

Expected: (an instance of java.lang.OutOfMemoryError and exception with cause expects type <class java.lang.OutOfMemoryError> with message "message")
     but: an instance of java.lang.OutOfMemoryError <com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!> is a com.hazelcast.core.HazelcastInstanceNotActiveException
Stacktrace was: com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:719)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.tryWrapInSameClass(AbstractInvocationFuture.java:1935)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.wrapRuntimeException(AbstractInvocationFuture.java:1911)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.wrapOrPeel(AbstractInvocationFuture.java:1892)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$ExceptionalResult.wrapForJoinInternal(AbstractInvocationFuture.java:1404)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.resolveAndThrowForJoinInternal(AbstractInvocationFuture.java:602)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.joinInternal(AbstractInvocationFuture.java:578)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation_ExceptionTest.waitForFuture(Invocation_ExceptionTest.java:163)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation_ExceptionTest.test(Invocation_ExceptionTest.java:151)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at com.hazelcast.spi.impl.operationservice.impl.InvocationRegistry.shutdown(InvocationRegistry.java:198)
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.shutdownInvocations(OperationServiceImpl.java:506)
	at com.hazelcast.spi.impl.NodeEngineImpl.shutdown(NodeEngineImpl.java:498)
	at com.hazelcast.instance.impl.Node.shutdownServices(Node.java:573)
	at com.hazelcast.instance.impl.Node.shutdown(Node.java:504)
	at com.hazelcast.instance.impl.OutOfMemoryHandlerHelper.tryShutdown(OutOfMemoryHandlerHelper.java:60)
	at com.hazelcast.instance.impl.DefaultOutOfMemoryHandler.onOutOfMemory(DefaultOutOfMemoryHandler.java:84)
	at com.hazelcast.instance.impl.DefaultOutOfMemoryHandler$MockitoMock$685448655.onOutOfMemory$accessor$m5DpzxYy(Unknown Source)
	at com.hazelcast.instance.impl.DefaultOutOfMemoryHandler$MockitoMock$685448655$auxiliary$4KoPnPHU.call(Unknown Source)
	at org.mockito.internal.invocation.RealMethod$FromCallable$1.call(RealMethod.java:40)
	at org.mockito.internal.invocation.RealMethod$FromBehavior.invoke(RealMethod.java:62)
	at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:127)
	at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:43)
	at org.mockito.Answers.answer(Answers.java:100)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:104)
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:63)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptSuperCallable(MockMethodInterceptor.java:110)
	at com.hazelcast.instance.impl.DefaultOutOfMemoryHandler$MockitoMock$685448655.onOutOfMemory(Unknown Source)
	at com.hazelcast.instance.impl.OutOfMemoryErrorDispatcher.onOutOfMemory(OutOfMemoryErrorDispatcher.java:187)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.handleOperationError(OperationRunnerImpl.java:357)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:205)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:147)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:125)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110)

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.junit.rules.ExpectedException.handleException(ExpectedException.java:252)
	at org.junit.rules.ExpectedException.access$000(ExpectedException.java:106)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:241)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:49)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:162)
	at java.base/java.lang.Thread.run(Thread.java:830)
```

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_ExceptionTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/Invocation_ExceptionTest.java
@@ -69,12 +69,12 @@ public class Invocation_ExceptionTest extends HazelcastTestSupport {
                 // thrown with a local stack trace; this test is about verifying the exception remains unwrapped
                 new Object[] {JOIN_INTERNAL, new CancellationException("message"), CancellationException.class,
                               IsNull.nullValue(Throwable.class)},
-                // CheckedException is wrapped in HazelcastException
+                // Checked exception is wrapped in HazelcastException
                 new Object[] {JOIN_INTERNAL, new Exception("message"), HazelcastException.class,
                               new RootCauseMatcher(Exception.class, "message")},
-                // OOME is wrapped in OOME
-                new Object[] {JOIN_INTERNAL, new OutOfMemoryError("message"), OutOfMemoryError.class,
-                              new RootCauseMatcher(OutOfMemoryError.class, "message")},
+                // Error subclass is wrapped in instance of same class
+                new Object[] {JOIN_INTERNAL, new ExceptionInInitializerError("message"), ExceptionInInitializerError.class,
+                              new RootCauseMatcher(ExceptionInInitializerError.class, "message")},
 
                 //// join()
                 // RuntimeException with a constructor accepting a Throwable cause
@@ -83,20 +83,18 @@ public class Invocation_ExceptionTest extends HazelcastTestSupport {
                 // RuntimeException with no constructor accepting a Throwable cause
                 new Object[] {JOIN, new IllegalThreadStateException("message"), CompletionException.class,
                               new RootCauseMatcher(IllegalThreadStateException.class, "message")},
-                // OperationTimeoutException: OperationTimeoutException is only expected to be
-                // thrown with a local stack trace; this test is about verifying the exception remains unwrapped
+                // OperationTimeoutException is wrapped in CompletionException
                 new Object[] {JOIN, new OperationTimeoutException("message"), CompletionException.class,
                               new RootCauseMatcher(OperationTimeoutException.class, "message")},
-                // CancellationException: CancellationException is only expected to be
-                // thrown with a local stack trace; this test is about verifying the exception remains unwrapped
+                // CancellationException is expected to be thrown from join() unwrapped
                 new Object[] {JOIN, new CancellationException("message"), CancellationException.class,
                               IsNull.nullValue(Throwable.class)},
-                // CheckedException is wrapped in HazelcastException
+                // Checked exception is wrapped in CompletionException
                 new Object[] {JOIN, new Exception("message"), CompletionException.class,
                               new RootCauseMatcher(Exception.class, "message")},
-                // OOME is wrapped in OOME
-                new Object[] {JOIN, new OutOfMemoryError("message"), CompletionException.class,
-                              new RootCauseMatcher(OutOfMemoryError.class, "message")},
+                // Error subclass is wrapped in CompletionException
+                new Object[] {JOIN, new ExceptionInInitializerError("message"), CompletionException.class,
+                              new RootCauseMatcher(ExceptionInInitializerError.class, "message")},
 
                 //// get()
                 // RuntimeException with a constructor accepting a Throwable cause
@@ -105,20 +103,18 @@ public class Invocation_ExceptionTest extends HazelcastTestSupport {
                 // RuntimeException with no constructor accepting a Throwable cause
                 new Object[] {GET, new IllegalThreadStateException("message"), ExecutionException.class,
                               new RootCauseMatcher(IllegalThreadStateException.class, "message")},
-                // OperationTimeoutException: OperationTimeoutException is only expected to be
-                // thrown with a local stack trace; this test is about verifying the exception remains unwrapped
+                // OperationTimeoutException is wrapped in ExecutionException
                 new Object[] {GET, new OperationTimeoutException("message"), ExecutionException.class,
                               new RootCauseMatcher(OperationTimeoutException.class, "message")},
-                // CancellationException: CancellationException is only expected to be
-                // thrown with a local stack trace; this test is about verifying the exception remains unwrapped
+                // CancellationException is expected to be thrown from get() unwrapped
                 new Object[] {GET, new CancellationException("message"), CancellationException.class,
                               IsNull.nullValue(Throwable.class)},
-                // CheckedException is wrapped in HazelcastException
+                // Checked exception is wrapped in HazelcastException
                 new Object[] {GET, new Exception("message"), ExecutionException.class,
                               new RootCauseMatcher(Exception.class, "message")},
-                // OOME is wrapped in OOME
-                new Object[] {GET, new OutOfMemoryError("message"), ExecutionException.class,
-                              new RootCauseMatcher(OutOfMemoryError.class, "message")},
+                // Error subclass is wrapped in ExecutionException
+                new Object[] {GET, new ExceptionInInitializerError("message"), ExecutionException.class,
+                              new RootCauseMatcher(ExceptionInInitializerError.class, "message")},
 
         };
     }

